c-scape/src/termios_/mod.rs

```diff
@@ -545,3 +545,52 @@ unsafe extern "C" fn tcflow(fd: c_int, action: c_int) -> c_int {
         None => -1,
     }
 }
+
+#[no_mangle]
+static mut pts_buffer: [c_char; 30] = [0; 30];
+
+#[no_mangle]
+unsafe extern "C" fn ptsname(fd: c_int) -> *mut c_char {
+    if ptsname_r(fd, pts_buffer.as_mut_ptr(), pts_buffer.len()) != 0 {
+        core::ptr::null_mut()
+    } else {
+        pts_buffer.as_mut_ptr()
+    }
+}
+
+#[no_mangle]
+unsafe extern "C" fn ptsname_r(fd: c_int, buf: *mut c_char, buflen: libc::size_t) -> c_int {
+    libc!(libc::ptsname_r(fd, buf, buflen));
+    if buf.is_null() {
+        set_errno(Errno(libc::EINVAL));
+        -1
+    } else {
+        let fd = BorrowedFd::borrow_raw(fd);
+        if let Ok(name) = rustix::pty::ptsname(fd, []) {
```

Comment:
Ah, I misread the `ptsname_r` documentation. I searched for errno and found these words:  "-1  is returned  on  error, with errno set to indicate the error" and missed that they were preceded by "A version of this function is documented on Tru64 and HP-UX, but on those implementations,". So you're right, this shouldn't use `convert_res`. It should `match` on the result returned from `rustix::pty::ptsname` and do `Err(err) => return err`.

Reply:
As per https://www.gnu.org/software/gnulib/manual/html_node/ptsname_005fr.html now, returning the raw eror number instead of setting `errno` : had the same confusion about returning -1